src/pip/_internal/wheel.py

```diff
@@ -818,9 +818,11 @@ def _build_one_inside_env(self, req, output_dir, python_tag=None):
                 builder = self._build_one_pep517
             else:
                 builder = self._build_one_legacy
-            if builder(req, temp_dir.path, python_tag=python_tag):
+
+            built_wheel = builder(req, temp_dir.path, python_tag=python_tag)
+            if built_wheel is not None:
                 try:
-                    wheel_name = os.listdir(temp_dir.path)[0]
+                    wheel_name = os.path.basename(built_wheel)
                     wheel_path = os.path.join(output_dir, wheel_name)
                     shutil.move(
                         os.path.join(temp_dir.path, wheel_name), wheel_path
```

Comment:
Since you're keeping the same filename, it seems like the code above could be simplified just by passing the `output_dir` to `shutil.move()`.

Reply:
Nice!